src/main/java/com/hivemq/persistence/SingleWriterService.java

```diff
@@ -208,15 +213,34 @@ public AtomicInteger getRunningThreadsCount() {
         return runningThreadsCount;
     }
 
-    public ExecutorService getExecutorService() {
-        return executorService;
-    }
-
     @NotNull
     public ExecutorService[] getCallbackExecutors() {
         return callbackExecutors;
     }
 
+    public void stop() {
+        final long start = System.currentTimeMillis();
+        if (log.isTraceEnabled()) {
```

Comment:
this if is then unnecessary

Reply:
This was just moved. I do not think we should differ the code bases to much.